File: sdk-java/src/main/java/io/littlehorse/sdk/common/auth/OAuthConfig.java
package io.littlehorse.sdk.common.auth;

import java.net.URI;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;

/**
 * Immutable OAuth client configuration.
 */
@Getter
@Builder
@EqualsAndHashCode
public class OAuthConfig {

    private final String clientId;
    private final String clientSecret;
    private final URI tokenEndpointURI;
}

File: sdk-dotnet/LittleHorseSDK/LittleHorse/Common/Authentication/Model/OAuthConfig.cs
namespace LittleHorse.Common.Authentication.Model
{
    public class OAuthConfig
    {
        public string ClientId { get; init; }
        public string ClientSecret { get; init; }
        public Uri TokenEndpointURI { get; init; }

        public OAuthConfig(string? clientId, string? clientSecret, string? tokenEndpointUrl)
        {
            if (string.IsNullOrEmpty(clientId) || string.IsNullOrEmpty(clientSecret) || string.IsNullOrEmpty(tokenEndpointUrl))
            {
                throw new ArgumentNullException("OAuth configuration is missing.");
            }

            ClientId = clientId;
            ClientSecret = clientSecret;
            TokenEndpointURI = new Uri(tokenEndpointUrl);
        }
    }
}
